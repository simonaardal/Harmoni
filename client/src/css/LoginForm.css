#LoginFormDiv{
        margin: auto;
        text-align: center;
        padding-top: 80px;
        background-image: url('../img/Backgrounds/Background 1.jpeg');
        background-size: cover;
        z-index: 2000;
        max-width: 100%;
}
#LoginFormWithoutFooter{
    min-height: 100vh;
    max-width: 100%;
}

#LoginFormForm{
    margin: auto;
    width: 50%;
    box-shadow: 0 0 0 0 rgb(0, 0, 0), 5px 5px 10px 0 rgb(0, 0, 0);
}
#loginFormForgotPW a{
    text-decoration: none;
    color: black;
}
#loginFormForgotPW{
    margin-top: 10px;
}

.LoginFormCard{
    /*background-image: url('../img/Backgrounds/Background 2.png');*/
    background-color: white;
    padding: 20px;
}

.LoginFormCard button{
    margin-top: 20px;
}

.LoginFormCard label {
    color: black;
}

#login-title{
    margin-bottom: 30px;
    color: black;
}

.login-button{
    width: 30%
}

#LoginFormButtons button{
    margin: 10px;
}

@Media only screen and (max-width: 780px){
    #LoginFormForm{
        margin: auto;
        width: 94%;
        box-shadow: 0 0 0 0 rgb(0, 0, 0), 5px 5px 10px 0 rgb(0, 0, 0);
    }
    #LoginFormButtons{
        width: 95%;
        margin: auto;
        display: grid;
        grid-template-columns: 1fr 1fr;
        text-align: center;
    }
    #LoginFormButtons button{
        display: grid;
        grid-template-columns: 1fr 1fr;
        margin: 10px;
    }
    .login-button{
        width: 95%;
        margin: 10px;
    }

}