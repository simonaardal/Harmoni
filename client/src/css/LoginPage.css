#LoginPageTitle{
    color: white;
}

#LoginPageTitle:hover{
    text-decoration: none;
    color: white;
}