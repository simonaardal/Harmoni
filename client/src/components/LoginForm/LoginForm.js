import React, {Component} from 'react';
import "../../css/LoginForm.css";
import UserService from "../../service/UserService.js"
import {authenticate} from "../../service/auth";
import {NavbarMainPage} from '../Navbar/Navbar.js'
import {toast} from "react-toastify";

let crypto = require('crypto');

/**
 * LoginForm
 */
class LoginForm extends Component{
    /**
     * @param {json} props 
     */
    constructor(props){
        super(props);
        this.state = {
            email: "",
            pw: "",
        };
        this.keyPressed = this.keyPressed.bind(this);
    }

    /**
     * Alerts the user if e-mail or password is wrong. Also gives an error if the user is not yet validated.
     * @function
     */
    notifyFailure = () => toast("E-post/passord var feil eller sÃ¥ er bruker ikke godkjent", {type: toast.TYPE.ERROR, position: toast.POSITION.BOTTOM_LEFT});



    /**
     * Submitting the values in state to a validate function to check if email/pw are valid.
     * If so, send the user to the home page/overview page.
     * @function
     */
    submit = () => {
        let userService = new UserService();
        userService.validate(this.state.email, this.state.pw)
            .then((response) => {
                let token = response.data.jwt;
                window.localStorage.setItem("token", token);
                window.location.hash = "/overview";
            })
            .then(authenticate)
            .catch((err) => {
                this.clearPassword();
                this.notifyFailure();
                console.error(err);
            })

    };

    /** 
     * Runs every time input-fields are updated. Updates the state with the most current values. 
     * @function
     */
    updateInputValue = (e) => {
        this.setState({
            [e.target.name]: e.target.value 
        });
    }

    /** 
     * Clears password field if an error is encountered. Mainly to discourage trial and error of passwords.
     * @function
     */
    clearPassword = () => {
        document.getElementById("password-input").value = "";
    }

    /** 
     * Submits the entered data when the ENTER key is pressed
     * @function
     */
    keyPressed(event){
        if (event.key === "Enter") {
            this.submit();
        }
    }

    /**
     * Displays a form for login. Contains a text-input-field for both e-mail and password and buttons for registration and login. If login is successful, the user is then redirected to the overview page.
     */
    render(){        
        return(
            <div>
                <form id="LoginFormForm">
                    <div className="card LoginFormCard">
                        <div className="card-body">
                            <h1 id="login-title">Logg inn</h1>
                            <div className="form-group" id="email-input-container">
                                <label htmlFor="email-input">E-post</label>
                                <input type="email" name={"email"} className="form-control" id="email-input" placeholder="Skriv inn e-post" name="email" onChange={this.updateInputValue} onKeyPress={this.keyPressed}/>
                            </div>
                            <div className="form-group" id="password-input-container">
                                <label htmlFor="password-input">Passord</label>
                                <input type="password" className="form-control" id="password-input" placeholder="Skriv inn passord" onKeyPress={this.keyPressed} name="pw" onChange={this.updateInputValue} />
                            </div>
                            <div id="LoginFormButtons">
                                <button type="button" className="btn btn-outline-dark login-button" onClick={this.submit}>Logg inn</button>
                                <button type="button" className="btn btn-outline-dark login-button" onClick={() => window.location.href="#/register"}>Registrer</button>
                            </div>
                            <div id="loginFormForgotPW">
                                <a href="#/forgotpassword">Glemt passord?</a>
                            </div>
                            </div>
                    </div>
                </form>
            </div>
        )
    }
}

export default LoginForm;