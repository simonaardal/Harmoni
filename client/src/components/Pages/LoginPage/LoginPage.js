import React, {Component} from 'react';

import '../../../css/LoginPage.css'
import LoginForm from "../../LoginForm/LoginForm"
import {NavbarMainPage} from '../../Navbar/Navbar'
import Footer from '../../Footer/Footer'
import {FooterTransparent} from '../../Footer/Footer'
import { auth } from "../../../service/auth.js"
import {Redirect} from 'react-router-dom';

/**
* @class LoginPage
* This component is the loginscreen for the host-portal. 
*/
class LoginPage extends Component{

    /** 
    * The component will now render on top of the screen.
    */
    componentDidMount(){
        window.scrollTo(0,0);
    }


    /** 
    * The component renders with the LoginForm-component and Footer at the bottom of the screen.
    */
    render() {
        return (
            <div className="pageSetup">
                <div id="LoginFormDiv">
                    <div id="LoginFormWithoutFooter">
                        <div id="MainPageTitle"><a id="LoginPageTitle" href="#/portal">HARMONI</a></div>
                        <LoginForm />
                    </div>
                    <div>
                        <FooterTransparent />
                    </div>    
                </div>
            </div> 
        );
    }
}

export default LoginPage;

/* <LoginForm />*/