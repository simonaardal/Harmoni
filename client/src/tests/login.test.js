import React from "react"
import ReactDOM from "react-dom";
import renderer from "react-test-renderer"
import LoginForm from "../components/LoginForm/LoginForm.js" 
import LoginPage from "../components/Pages/LoginPage/LoginPage.js"

test("LoginPage renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<LoginPage/>, div);
    ReactDOM.unmountComponentAtNode(div);
});

describe("Login-form component", () => {
    test.skip("Matches the snapshot", () => {
        const form = renderer.create(<LoginForm />);
        expect(form.toJSON()).toMatchSnapshot(); 
    });
});