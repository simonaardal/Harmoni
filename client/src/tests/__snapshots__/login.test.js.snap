// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login-form component Matches the snapshot 1`] = `
<form>
  <h1
    id="login-title"
  >
    Innlogging
  </h1>
  <div
    class="form-group"
    id="email-input-container"
  >
    <label
      for="email-input"
    >
      E-post
    </label>
    <input
      class="form-control"
      id="email-input"
      name="email"
      onChange={[Function]}
      placeholder="Skriv inn e-post"
      type="email"
    />
  </div>
  <div
    class="form-group"
    id="password-input-container"
  >
    <label
      for="password-input"
    >
      Passord
    </label>
    <input
      class="form-control"
      id="password-input"
      name="pw"
      onChange={[Function]}
      placeholder="Skriv inn passord"
      type="password"
    />
  </div>
  <button
    class="btn btn-primary"
    id="login-button"
    onClick={[Function]}
    type="button"
  >
    Logg inn
  </button>
</form>
`;
